Testlio/testlio-jenkins-plugin
# GREEN-705: Implemented plugin UI and data validation. Defined data structure.

#### JIRA
[GREEN-705](https://testlions.atlassian.net/browse/GREEN-705)

#### Related PRs
- none

#### Background and Solution
- Implemented plugin configuration page
- Implemented run configuration page
- Added input fields validation
- Defined data structure

#### Testing Strategy
1. Deploy local Jenkins instance using command `mvn hpi:run -Dport=5002`
2. Navigate to http://localhost:5002/jenkins/
3. If necessary, create a new Freestyle project
4. Go to project configuration (on project page click `Configure` button on side panel)
5. Click `Add build step`
6. Select `Schedule automated run on Testlio`
7. Pass run configuration parameters (check that configuration form works as expected)
8. Click `Save`
9. On project page's side panel click `Build Now`
10. Navigate to appeared build page
11. Open `Console Log` and ensure that object with valid run configuration parameters printed
12. Open `Run Configuration` and ensure that page with run configuration params properly rendered

#### This PR Has Been Tested With the Following Browsers
- [x] Chrome (Mac OS)
- [ ] Chrome (Windows)
- [ ] Safari (Mac OS)
- [ ] Firefox (Mac OS)

#### Are the Changes Covered With Tests?
